esp/esdllib/esdl_script.cpp

```diff
@@ -136,26 +141,133 @@ class CEsdlTransformOperationParameter : public CEsdlTransformOperationVariable
+    {
+        if (m_traceName.isEmpty())
+            m_traceName.set(tree->queryProp("@name"));
+        if (!isEmptyString(attrName))
+            m_name.set(attrName);
+        else
+        {
+            m_name.set(tree->queryProp("@name"));
+
+            const char *xpath_name = tree->queryProp("@xpath_name");
+            if (!isEmptyString(xpath_name))
+                m_xpath_name.setown(compileXpath(xpath_name));
+            else if (m_name.isEmpty())
+                esdlOperationError(ESDL_SCRIPT_MissingOperationAttr, m_tagname, "without name", m_traceName, !m_ignoreCodingErrors); //don't mention value, it's deprecated
+        }
+
+        const char *select = tree->queryProp("@select");
+        if (isEmptyString(select))
+            esdlOperationError(ESDL_SCRIPT_MissingOperationAttr, m_tagname, "without select", m_traceName, !m_ignoreCodingErrors); //don't mention value, it's deprecated
+        m_select.setown(compileXpath(select));
+    }
+
+    virtual void toDBGLog() override
+    {
+#if defined(_DEBUG)
+        DBGLOG(">%s> %s(name(%s), select('%s'))", m_traceName.str(), m_tagname.str(), (m_xpath_name) ? m_xpath_name->getXpath() : m_name.str(), m_select->getXpath());
+#endif
+    }
+
+    virtual ~CEsdlTransformOperationSetSectionAttributeBase(){}
+
+    virtual const char *getSectionName() = 0;
+
+    virtual bool process(IEsdlScriptContext * context, IPropertyTree *content, IXpathContext * xpathContext) override
+    {
+        if ((!m_name && !m_xpath_name) || !m_select)
+            return false; //only here if "optional" backward compatible support for now (optional syntax errors aren't actually helpful)
+        try
+        {
+            StringBuffer name;
+            if (m_xpath_name)
+                xpathContext->evaluateAsString(m_xpath_name, name);
+            else
+                name.set(m_name);
+
+            StringBuffer value;
+            xpathContext->evaluateAsString(m_select, value);
+            context->setAttribute(getSectionName(), name, value);
+        }
+        catch (IException* e)
+        {
+            int code = e->errorCode();
+            StringBuffer msg;
+            e->errorMessage(msg);
+            e->Release();
+            esdlOperationError(code, m_tagname, msg, m_traceName, !m_ignoreCodingErrors);
+        }
+        catch (...)
+        {
+            esdlOperationError(ESDL_SCRIPT_Error, m_tagname, "unknown exception processing", m_traceName, !m_ignoreCodingErrors);
+        }
+        return false;
+    }
+};
+
+class CEsdlTransformOperationStoreValue : public CEsdlTransformOperationSetSectionAttributeBase
+{
+public:
+    CEsdlTransformOperationStoreValue(IPropertyTree *tree, const StringBuffer &prefix) : CEsdlTransformOperationSetSectionAttributeBase(tree, prefix, nullptr)
+    {
+    }
+
+    virtual ~CEsdlTransformOperationStoreValue(){}
+    const char *getSectionName() override {return "store";}
```

Comment:
The string literals should be defined names. Define the convention now, and hopefully avoid inconsistencies with future extensions.

Reply:
Ok agreed.